File: 1.0/src/cz/mg/vulkan/PFNvkDestroyDescriptorSetLayout.java
package cz.mg.vulkan;

public class PFNvkDestroyDescriptorSetLayout extends VkFunctionPointer {
    public PFNvkDestroyDescriptorSetLayout() {
    }

    protected PFNvkDestroyDescriptorSetLayout(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkDestroyDescriptorSetLayout(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyDescriptorSetLayout(long value) {
        setValue(value);
    }

    public PFNvkDestroyDescriptorSetLayout(VkInstance instance) {
        super(instance, new VkString("vkDestroyDescriptorSetLayout"));
    }

    public void call(VkDevice device, VkDescriptorSetLayout descriptorSetLayout, VkAllocationCallbacks pAllocator){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), descriptorSetLayout != null ? descriptorSetLayout.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long descriptorSetLayout, long pAllocator);



}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkDestroyDescriptorSetLayout.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyDescriptorSetLayout_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong descriptorSetLayout, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyDescriptorSetLayout f = (PFN_vkDestroyDescriptorSetLayout)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkDescriptorSetLayout*)jniLongToPointer(descriptorSetLayout)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

